test-suite/src/function/ifnull.rs

```diff
@@ -0,0 +1,74 @@
+use crate::*;
+
+test_case!(ifnull, async move {
+    use chrono::{NaiveDate, NaiveDateTime, NaiveTime};
+    use gluesql_core::{executor::Payload, prelude::Value::*};
+    use rust_decimal::Decimal;
+    let test_cases = vec![
+        (
+            r#"CREATE TABLE SingleItem (id integer null, int8 int(8) null, dec decimal null, 
+                                        dt date null, mystring Text null,
+                                        mybool Boolean null, myfloat float null,
+                                        mytime time null, mytimestamp timestamp null)"#,
+            Payload::Create,
+        ),
+        (
+            r#"INSERT INTO SingleItem VALUES (0, 1, 2, "2022-05-23", "this is a string", true, 3.15,
+                          "01:02:03", "1970-01-01 00:00:00 -00:00")"#,
+            Payload::Insert(1),
+        ),
+        (
+            r#"INSERT INTO SingleItem VALUES (null, null, null, null, null, null, null, null, null)"#,
+            Payload::Insert(1),
+        ),
+        (
+            r#"SELECT ifnull(id, 1) as ID, ifnull(int8, 2) as int8, ifnull(dec, 3) 
+            FROM SingleItem where id is not null"#,
+            select!("ID" | "int8" | "ifnull(dec, 3)"; I64 | I8 | Decimal; 0 1 Decimal::from(2i8)),
+        ),
+        (
+            // notice that this example returns I64 I64 I64, where the previous example returned I64 I8 Decimal.
+            // is this behavior desirable?  see https://dev.mysql.com/doc/refman/5.7/en/flow-control-functions.html#function_ifnull
+            r#"SELECT ifnull(id, 1) as ID, ifnull(int8, 2) as int8, ifnull(dec, 3) 
```

Comment:
OK..  sounds good to me,   since IFNULL can return some other data type such as IFNULL(myint, "my string")  then it makes sense to leave it this way..